Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_hidden.nut

```diff
@@ -153,35 +159,62 @@ void function RemoveHidden()
 	}
 }
 
-void function PredatorMain()
+void function PredatorMain( entity player )
 {
+	float playerVel
+
 	while (true) 
 	{
 		WaitFrame()
 		if(!IsLobby())
 		{
-			foreach (entity player in GetPlayerArray())
+			if ( !IsValid(player) || !IsAlive(player) || player.GetTeam() != TEAM_IMC )
+				continue
+
+			vector playerVelV = player.GetVelocity()
+			playerVel = sqrt(playerVelV.x * playerVelV.x + playerVelV.y * playerVelV.y + playerVelV.z * playerVelV.z)
+
+			if (playerVel/300 < 1.3)
 			{
-				if (player == null || !IsValid(player) || !IsAlive(player) || player.GetTeam() != TEAM_IMC)
-					continue
-				vector playerVelV = player.GetVelocity()
-				float playerVel
-				playerVel = sqrt(playerVelV.x * playerVelV.x + playerVelV.y * playerVelV.y + playerVelV.z * playerVelV.z)
-				float playerVelNormal = playerVel * 0.068544
-				if (playerVel/300 < 1.3)
+				player.SetCloakFlicker(0, 0)
+				player.kv.VisibilityFlags = 0
+				wait 0.5
+				if (file.isVisible)
 				{
-					player.SetCloakFlicker(0, 0)
-					player.kv.VisibilityFlags = 0
-				}
-				else
-				{
-					player.SetCloakFlicker(0.2 , 1 )
-					player.kv.VisibilityFlags = 0
-					float waittime = RandomFloat(0.5)
-					wait waittime
-					player.kv.VisibilityFlags = ENTITY_VISIBLE_TO_EVERYONE
+					file.isVisible = false
+					player.Signal("VisibleNotification")
 				}
 			}
+			else
+			{
+				player.SetCloakFlicker(0.2 , 1 )
+				player.kv.VisibilityFlags = 0
+				float waittime = RandomFloat(0.5)
+				wait waittime

```

Comment:
when a player disconnects during this wait the server crashes

Reply:
Added EndSignals so threads will end if player disconnects

Reply:
Confirmed fixed with f460f4aceb27cd5767c3e6fdce91434c6a19ba76